datadog_checks_base/datadog_checks/checks/openmetrics/mixins.py

```diff
@@ -641,113 +453,83 @@ def _submit(self, metric_name, message, scraper_config, hostname=None):
+                self.gauge("{}.{}.count".format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=custom_hostname)
 
     def _submit_gauges_from_histogram(self, metric_name, metric, scraper_config, hostname=None):
         """
         Extracts metrics from a prometheus histogram and sends them as gauges
         """
-        # histograms do not have a value attribute
-        val = getattr(metric, self.METRIC_TYPES[4]).sample_count
-        if self._is_value_valid(val):
-            tags = self._metric_tags(metric_name, val, metric, scraper_config, hostname=hostname)
-            self.gauge('{}.{}.count'.format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=hostname)
-        else:
-            self.log.debug("Metric value is not supported for metric {}.count.".format(metric_name))
-        val = getattr(metric, self.METRIC_TYPES[4]).sample_sum
-        if self._is_value_valid(val):
-            tags = self._metric_tags(metric_name, val, metric, scraper_config, hostname=hostname)
-            self.gauge('{}.{}.sum'.format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=hostname)
-        else:
-            self.log.debug("Metric value is not supported for metric {}.sum.".format(metric_name))
-        if scraper_config['send_histograms_buckets']:
-            for bucket in getattr(metric, self.METRIC_TYPES[4]).bucket:
-                val = bucket.cumulative_count
-                limit = bucket.upper_bound
-                if self._is_value_valid(val):
-                    tags = self._metric_tags(metric_name, val, metric, scraper_config, hostname=hostname) + ['upper_bound:{}'.format(limit)]
-                    self.gauge('{}.{}.count'.format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=hostname)
-                else:
-                    self.log.debug("Metric value is not supported for metric {}.count.".format(metric_name))
-
-    def _metric_tags(self, metric_name, val, metric, scraper_config, hostname=None):
+        for sample in metric.samples:
+            val = sample[self.SAMPLE_VALUE]
+            if not self._is_value_valid(val):
+                self.log.debug("Metric value is not supported for metric {}".format(sample[self.SAMPLE_NAME]))
+                continue
+            custom_hostname = self._get_hostname(hostname, sample, scraper_config)
+            tags = self._metric_tags(metric_name, val, sample, scraper_config, hostname)
+            if sample[self.SAMPLE_NAME].endswith("_sum"):
+                self.gauge("{}.{}.sum".format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=custom_hostname)
+            elif sample[self.SAMPLE_NAME].endswith("_count"):
+                self.gauge("{}.{}.count".format(scraper_config['namespace'], metric_name), val, tags=tags, hostname=custom_hostname)
+            elif sample[self.SAMPLE_NAME].endswith("_bucket") and scraper_config['send_histograms_buckets']:
```

Comment:
We can probably just merge these two conditionals into:
`elif sample[self.SAMPLE_NAME].endswith("_count") or (sample[self.SAMPLE_NAME].endswith("_bucket") and scraper_config['send_histograms_buckets']):`